third_party/grpc_java/java_grpc_library.bzl

```diff
@@ -18,7 +18,7 @@ def _gensource_impl(ctx):
   srcdotjar = ctx.actions.declare_file(ctx.label.name + "_src.jar")
 
   srcs = [f for dep in ctx.attr.srcs for f in dep[ProtoInfo].direct_sources]
-  includes = [f for dep in ctx.attr.srcs for f in dep[ProtoInfo].transitive_imports]
+  includes = [f for dep in ctx.attr.srcs for f in dep[ProtoInfo].transitive_imports.to_list()]
```

Comment:
Ideally, you could avoid flattening the depset (https://docs.bazel.build/versions/master/skylark/performance.html#use-depsets) to reduce memory usage.

1. The inputs argument of actions.run can accept a depset.
2. For arguments, it's not as easy, but ctx.actions.args() should work.

I'd suggest to do at least 1. But in any case, this fix is enough for me as it fixes the issue.

Reply:
Challenge accepted. protobuf and grpc deps upgraded. Using the grpc-java version of java_grpc_library.bzl now, should have fewer issues with it falling out of date.